cni/pkg/nodeagent/informers.go

```diff
@@ -212,19 +216,42 @@ func (s *InformerHandlers) reconcilePod(input any) error {
 		wasAnnotated := oldPod.Annotations != nil && oldPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
 		isAnnotated := newPod.Annotations != nil && newPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
 		shouldBeEnabled := util.PodRedirectionEnabled(ns, newPod)
+		wasTerminated := kube.CheckPodTerminal(oldPod)
+		isTerminated := kube.CheckPodTerminal(newPod)
+		// only checks if the pod state just changed from not-terminated to terminated.
+		// if pod state changed in *any* other way, this should be false.
+		justTerminated := !wasTerminated && isTerminated
+
+		// Check intent (labels) versus status (annotation) - is there a delta we need to fix?
+		changeNeeded := (isAnnotated != shouldBeEnabled) && !isTerminated
+
+		// nolint: lll
+		log.Debugf("pod %s events: wasAnnotated(%v), isAnnotated(%v), shouldBeEnabled(%v), changeNeeded(%v), wasTerminated(%v), isTerminated(%v), oldPod(%+v), newPod(%+v)",
+			pod.Name, wasAnnotated, isAnnotated, shouldBeEnabled, changeNeeded, wasTerminated, isTerminated, oldPod.ObjectMeta, newPod.ObjectMeta)
+
+		// If it was a job pod that (a) we captured and (b) just terminated (successfully or otherwise)
+		// remove it (the pod process is gone, but kube will keep the Pods around in
+		// a terminated || failed state - we should still do cleanup)
+		//
+		// Note that kube may either restart the same pod, or spawn a new one, depending on how
```

Comment:
I don't think it can restart the same **pod**, just the containers of the pod.

Reply:
> Also, if it is going to, it will never transition to Failure or Success. a pod in those states will never change afaik

Yeah - that statement is at some variance with "the pod is not restarted but the containers are" tho. Or my mental model of how K8S works is broken.